src/book/03-guides/04-general/04-documentation.mdx

```diff
@@ -0,0 +1,78 @@
+---
+section: Guides
+chapter: General
+title: Documentation
+description: How to properly document code.
+slug: /guides/general/documentation
+---
+
+## Conventions
+
+We use the javadoc documentation style for comments that are able to be formatted with doxygen. Each variable, function and class declaration should have a comment detailing the information about it in the `.hpp` file.
+
+### Directives
+
+This is a list of directives that we use.
+
+- `@brief` A brief description of the following code. Everything should at least have this tag.
+- `@detail` A complete and detailed description of the code.
+- `@param` Details a paramter that is passed to a function.
+- `@return` Details what a function returns.
+- `@retval` Details specific return value meanings, usually for enums, booleans or error codes.
+- `@throws` Details on what a function throws.
+- `@tparam` Details on the template parameter, usually a [named requirement](https://en.cppreference.com/w/cpp/named_req) or other requirements that must be fulfilled.
+
+This is a list of extra things that may be used, see [doxygen](https://www.doxygen.nl/manual/commands.html#cmd_intro) documentation for the full list.
+
+- `@example <file-name>` An example of how the function or class should be used is at `<file-name>`.
+- `@see` Tells the reader to read documentation for another class, function or variable.
+- `@attention` Tells the reader that they must pay attention.
+- `@warning` Tells the reader something that may go wrong.
+- `@note` Tells the reader to take note.
+- `@remark` A remark about the thing.
+- `@todo` Adds a todo to the documentation.
+- `@post` Details a post condition.
+- `@pre` Details a pre condition.
+- `@code` and `@endcode` starts and finishes a code block, like \` and in NUbook.
+- `@copydoc` Copies the documentation from another thing.
+- `@dot` and `@enddot` Creates a dot graph.
+- `@msc` and `@endmsc` Creates a [mscgen](https://www.mcternan.me.uk/mscgen/) diagram.
+- `@startuml` and `@enduml` Creates a [uml](https://plantuml.com/) diagram.
+- `@f$`, `@f[`, `@f]`, `@f{<environment>}{` and `@f}` Creates latex mathematics.
+- `@verbatim` and `@endverbatim` Starts and finishes verbatim output.
+- `@author` The author of this code, this is not used often in NUbots as we generally rely on the git blame instead.
+
+Basic html can be used, like `<b>` and `<em>`.
+
+### Example
+
+This example is more than normally required, but tries to give an example of how most directives work.
+
+```c++
+/**
+ * @brief Is super cool.
```

Comment:
I appreciate the meme value of this example hahaha
but... I think it would be better if we had an example with better tags. Something which actually would exist might make it easier. I think that `@tparam U The second argument` is a poor canonical example

Reply:
Here's another example that could be used:

```cpp
/**
 * @brief Represents a robot in the simulation environment
 * @details As far as the simulation is concerned, robots have only two behaviours: move and make a sound. This class supports modelling both behaviours.
 * @tparam T The robot's type of motion (walk, roll, etc), should be a subtype of the #Motion class
 * @tparam U The type of sound the robot makes, should be a subtype of the #Sound class
 */
template <typename T, typename U>
class SimulationRobot {
    /// @brief The name of the robot, should be unique across all robots in the simulation
    private std::string name;

    /// @brief How loud the robot sounds, in decibels
    /// @see AudioEngine::playback() and Sound::default_volume
    private int sound_volume = 60;

    /**
     * @brief Check if the robot has moved since the given timestamp
     * @param timestamp The timestamp (in milliseconds of simulation time) to check for motion after
     * @throws NotFoundError The robot object was not found in the current simulation world
     * @return Whether or not the robot has moved since the timestamp
     * @retval true The robot has moved since the timestamp
     * @retval false The robot has not moved since the timestamp
     */
    [[nodiscard]] bool has_moved_since(const int& timestamp) const;
}
```

Reply:
I also think we should prefer specifying `@brief` and `@details` without the tags (use the default first line + subsequent lines behaviour).